nntrainer/layers/rnn.cpp

```diff
@@ -95,7 +96,38 @@ void RNNLayer::setProperty(const PropertyType type, const std::string &value) {
   }
 }
 
-void RNNLayer::forwarding(bool training) {}
+void RNNLayer::forwarding(bool training) {
+  Tensor &weight_xh =
+    weightAt(static_cast<int>(RNNParams::weight_xh)).getVariableRef();
+  Tensor &weight_hh =
+    weightAt(static_cast<int>(RNNParams::weight_hh)).getVariableRef();
+  Tensor &bias_h =
+    weightAt(static_cast<int>(RNNParams::bias_h)).getVariableRef();
+
+  Tensor &hidden_ = net_hidden[0]->getVariableRef();
+  Tensor &input_ = net_input[0]->getVariableRef();
+
+  for (unsigned int b = 0; b < input_dim[0].batch(); ++b) {
+    Tensor islice = input_.getBatchSlice(b, 1);
+    Tensor oslice = hidden_.getBatchSlice(b, 1);
+
+    for (unsigned int t = 0; t < islice.height(); ++t) {
+      Tensor xs = Tensor(TensorDim(1, 1, 1, islice.width()),
+                         islice.getAddress(t * islice.width()));
+      unsigned int id = 0;
+      if (t > 0) {
+        id = t - 1;
+      }
+      Tensor hs = Tensor(TensorDim(1, 1, 1, oslice.width()),
```

Comment:
Tensor(Tensordim, float * data) will create a new copy so I think this will not update hidden.

Could you make a small `model` test to verify the output?

Reply:
Right. I will update this.